File: ciris_sdk/exceptions.py
class CIRISError(Exception):
    """Base exception for the SDK."""

class CIRISAPIError(CIRISError):
    """API errors with status codes"""
    def __init__(self, status_code: int, message: str):
        super().__init__(f"API Error {status_code}: {message}")
        self.status_code = status_code
        self.message = message

class CIRISTimeoutError(CIRISError):
    """Timeout errors"""

class CIRISConnectionError(CIRISError):
    """Connection errors - triggers retry"""

File: CIRISGUI/apps/agui/lib/ciris-sdk/exceptions.ts
// CIRIS TypeScript SDK - Exception Classes

export class CIRISError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'CIRISError';
  }
}

export class CIRISAPIError extends CIRISError {
  constructor(
    public status: number,
    message: string,
    public detail?: string,
    public type?: string
  ) {
    super(message);
    this.name = 'CIRISAPIError';
  }
}

export class CIRISAuthError extends CIRISError {
  constructor(message: string) {
    super(message);
    this.name = 'CIRISAuthError';
  }
}

export class CIRISConnectionError extends CIRISError {
  constructor(message: string) {
    super(message);
    this.name = 'CIRISConnectionError';
  }
}

export class CIRISTimeoutError extends CIRISError {
  constructor(message: string) {
    super(message);
    this.name = 'CIRISTimeoutError';
  }
}

export class CIRISValidationError extends CIRISError {
  constructor(message: string) {
    super(message);
    this.name = 'CIRISValidationError';
  }
}

export class CIRISRateLimitError extends CIRISAPIError {
  constructor(
    public retryAfter: number,
    public limit: number,
    public window: string
  ) {
    super(429, `Rate limit exceeded. Retry after ${retryAfter} seconds`);
    this.name = 'CIRISRateLimitError';
  }
}